### PR TITLE
Splitted chain_ros.h into chain_ros.h and ros_chain.cpp

### PR DESCRIPTION
**Pull request by mistoll**
This patch moves a lot of code from the header to a cpp file.

### CODE REVIEW

**Comment by mathias-luedtke** [comment]
Thanks for the effort!
Only some minor changes are needed before I can merge it :)

File: canopen_chain_node/CMakeLists.txt [hunk 1]
```diff
@@ -106,19 +106,25 @@ include_directories(
...
+  src/ros_chain.cpp
+)
+target_link_libraries(canopen_ros_chain
+  ${Boost_LIBRARIES}
+  ${catkin_LIBRARIES}
+)
+
+## Add cmake target dependencies of the executable/library
+## as an example, message headers may need to be generated before nodes
+add_dependencies(canopen_ros_chain cob_srvs_gencpp) # TODO: workaround needed?
```
Comment (mathias-luedtke) [inline canopen_chain_node/CMakeLists.txt]: This one slipped through: can you please change this to `add_dependencies(canopen_ros_chain ${catkin_EXPORTED_TARGETS})`

File: canopen_chain_node/CMakeLists.txt [hunk 2]
```diff
@@ -106,19 +106,25 @@ include_directories(
 )
 
 ## Declare a cpp library
-# add_library(canopen_chain_node
-#   src/${PROJECT_NAME}/canopen_chain_node.cpp
-# )
+add_library(canopen_ros_chain
```
Comment (mathias-luedtke) [inline canopen_chain_node/CMakeLists.txt:109]: Please add ths target to the install rules, too.

**Comment by mistoll** [comment]
Not sure if I got the install right, didn't use that, yet.

**Comment by mathias-luedtke** [comment]
Looks fine, the node starts up.
Let's see if Jenkins is happy as well :)

**Pull request merged**